### DISCUSSION TITLE
MapperIgnore attributes not being generated on IQueryable

### DISCUSSION THREAD

**Discussion by haydenoap** [riok/mapperly#842] [Q&A]
**Describe the bug**
When creating a new mapper class, if you only have an IQueryable mapping method, the MapperIgnore attributes are not honoured and will continue to generate the properties. If you add the same mapping method without the non-IQueryable, it will properly generate both the IQueryable and non-IQueryable ignored properties.

**To Reproduce**
Steps to reproduce the behaviour:
1. Create a new mapper class
2. Add a mapping method `IQueryable<CarDto> ProjectToDto(this IQueryable<Car> car);`
3. Add an MapperIgnoreTarget attribute on the Tires property `[MapperIgnoreTarget(nameof(CarDto.Tires))]`
3a. The generated code will still have the Tires property
4. Add another mapping method `CarDto ToDto(this Car car);`
5. Add the same MapperIgnoreTarget attribute on the new mapping method `[MapperIgnoreTarget(nameof(CarDto.Tires))]`
5a. The generated code will correctly ignore the Tires property on both mapping methods

**Expected behavior**
At 3a, the generated code *should* ignore the Tires property regardless of whether a non-IQueryable mapping method is present or not.

**Code snippets**
The models and dtos are the same as the ones provided [in the docs](https://mapperly.riok.app/docs/getting-started/generated-mapper-example/). The mapping class is as followed as of step 3a:

```csharp
[Mapper]
public static partial class CarMapper
{
   [MapperIgnoreTarget(nameof(CarDto.Tires))]
   public static partial IQueryable<CarDto> ProjectToDto(this IQueryable<Car> car);
}
```

Here is the mapping class as of step 5a:

```csharp
[Mapper]
public static partial class CarMapper
{
   [MapperIgnoreTarget(nameof(CarDto.Tires))]
   public static partial IQueryable<CarDto> ProjectToDto(this IQueryable<Car> car);

   [MapperIgnoreTarget(nameof(CarDto.Tires))]
   public static partial CarDto ToDto(this Car car);
}
```

**Environment:**
- Mapperly Version: 3.2.0
- .NET Version: 7.0.302
- Target Framework: net6.0
- IDE: Visual Studio Professional 2022 v17.6.2
- OS: Windows 10 Pro build 19045.3570

**Comment by latonz** [comment]
Mapperly attributes are alyways applied to the parameter / return type of mapping methods. This should work:
```c#
[Mapper]
public static partial class CarMapper
{
   public static partial IQueryable<CarDto> ProjectToDto(this IQueryable<Car> car);

   [MapperIgnoreTarget(nameof(CarDto.Tires))]
   private static partial CarDto ToDto(this Car car);
}
```

Mapperly uses the configuration of `ToDto` to build the mapping of the IQueryable.

**Reply by mpickers** [reply]
Perhaps updating the "Property Configuration" section to something like:

"Since Mapperly attributes are always applied to the exact parameter / return type of mapping methods, you will need to add a partial mapping method definition with attributes as needed to configure property mappings for the underlying types being mapped (not the IQueryable<>). These will then be applied to the IQueryable<> mapping version. Set these methods to private to hide them from callers."

may make it more explicit that this is how Mapperly works.
